### PR TITLE
Add support for Xiaomi Mijia STYJ02YM (viomi.vacuum.v7)

### LINKED ISSUE #550: Support for new vaccum Xiaomi Mijia STYJ02YM
Hello,

It's possible to add the new vaccum xaomi.
This model : https://fr.gearbest.com/vacuum-cleaners/pp_009487950353.html

Thanks

### PR DESCRIPTION
**Pull request by rytilahti**
Based on findings done by @nqkdev and contributions by @rumpeltux - thanks a lot for help!
This is the initial PR to fix #550 - plenty of values are still unknown so any help is welcome!

### CODE REVIEW

File: miio/viomivacuum.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+
+class ViomiVacuumSpeed(Enum):
+    Silent = 0
+    Standard = 1
+    Medium = 2
+    Turbo = 3
+
+
+class ViomiVacuumState(Enum):
```
Comment (rumpeltux) [inline miio/viomivacuum.py]: It seems to sometimes also return 0 for state (idle, not docked), subquent queries return ViomiVacuumState.Idle (1).

```
DEBUG:miio.device:192.168.178.82:54321 >>: {'id': 405, 'method': 'get_prop', 'params': ['run_state', 'mode', 'err_state', 'battary_life', 'box_type', 'mop_type', 's_time', 's_area', 'suction_grade', 'water_grade', 'remember_map', 'has_map', 'is_mop', 'has_newmap']}
DEBUG:miio.device:192.168.178.82:54321 (ts: 2019-11-23 09:13:34, id: 405) << {'result': [0, 0, 0, 76, 0, 0, 31, 18.13, 0, 12, 1, 1, 1, 0], 'id': 405}
ValueError: 0 is not a valid ViomiVacuumState
```

Comment (syssi) [reply miio/viomivacuum.py]: Fixed.

**Comment by rytilahti** [comment]
@rezmus @rumpeltux I just added some missing documentation & the carpetturbo setter. 

I think we can merge this as it is, new features can be added then later on as needed, or do you want to get some other features implemented before doing that?

**Comment by rytilahti** [comment]
On the other hand, let's just merge this. It's simple to add new features on top of it later on, thanks :+1:

**Pull request merged**